packages/react/src/components/MenuButton/MenuButton.tsx

```diff
@@ -0,0 +1,337 @@
+    return {
+      menuId: getOrGenerateIdFromShorthand('menubutton-menu-', props.menu, state.menuId, true),
+      triggerId: getOrGenerateIdFromShorthand(
+        'menubutton-trigger-',
+        props.trigger,
+        state.triggerId,
+        true,
+      ),
+    }
+  }
+
+  triggerRef = React.createRef<HTMLElement>()
+  menuRef = React.createRef<HTMLElement>()
+
+  actionHandlers = {
+    closeAndFocusNext: e => this.closeAndFocus(e, 'next'),
+    closeAndFocusPrevious: e => this.closeAndFocus(e, 'previous'),
+    openAndFocusFirst: e => this.openAndFocus(e, 'first'),
+    openAndFocusLast: e => this.openAndFocus(e, 'last'),
+  }
+
+  closeAndFocus(e: Event, which: 'next' | 'previous') {
+    const renderCallback = () => focusNearest(this.triggerRef.current, which)
+    this.trySetState(
+      {
+        open: false,
+        autoFocus: false,
+      },
+      renderCallback,
+    )
+    e.preventDefault()
+  }
+
+  openAndFocus(e: Event, which: 'first' | 'last') {
+    const renderCallback = () => focusMenuItem(this.menuRef.current, which)
+    this.trySetState(
+      {
+        open: true,
+        autoFocus: false, // focused by renderCallback
+      },
+      renderCallback,
+    )
+    e.preventDefault()
+  }
+
+  handleOpenChange = (e, { open }: PopupProps) => {
+    _.invoke(this.props, 'onOpenChange', e, { ...this.props, ...{ open } })
+    this.trySetState({
+      open,
+      autoFocus: true,
+    })
+  }
+
+  handleMenuItemClick = (predefinedProps?: MenuProps) => (
+    e: React.SyntheticEvent,
+    itemProps: MenuItemProps,
+  ) => {
+    _.invoke(predefinedProps, 'onItemClick', e, itemProps)
+    if (!itemProps || !itemProps.menu) {
+      // do not close if clicked on item with submenu
+      this.trySetState({ open: false, autoFocus: false })
+    }
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    unhandledProps,
+    accessibility,
+    styles,
+  }: RenderResultConfig<MenuButtonProps>): React.ReactNode {
+    const {
+      // MenuButton props:
+      contextMenu,
+      menu,
+      // Popup props:
+      accessibility: accessibilityProp,
+      align,
+      className,
+      defaultOpen,
+      inline,
+      mountDocument,
+      mountNode,
+      mouseLeaveDelay,
+      offset,
+      on,
+      onOpenChange,
+      open,
+      pointing,
+      position,
+      tabbableTrigger,
+      styles: stylesProp,
+      target,
+      trigger,
+      unstable_pinned,
+      variables,
+    } = this.props
+
+    const popupProps = _.omitBy(
+      {
+        accessibility: accessibilityProp,
+        align,
+        className,
+        defaultOpen,
+        inline,
+        mountDocument,
+        mountNode,
+        mouseLeaveDelay,
+        offset,
+        on,
+        onOpenChange,
+        open,
+        pointing,
+        position,
+        tabbableTrigger,
+        styles: stylesProp,
+        target,
+        trigger,
+        unstable_pinned,
+        variables,
+      },
+      _.isUndefined,
```

Comment:
@layershifter is this correct? I had to remove props with `undefined` value, otherwise the default props were overriden to `undefined` when not present.